cavern/src/main/java/org/opencadc/cavern/ServiceAvailability.java

```diff
@@ -150,35 +150,40 @@ public Availability getStatus() {
             
             URI credURI = null;
             try {
-                credURI = localAuthority.getServiceURI(Standards.CRED_PROXY_10.toString());
-                URL url = reg.getServiceURL(credURI, Standards.VOSI_AVAILABILITY, AuthMethod.ANON);
-                if (url != null) {
-                    CheckResource checkResource = new CheckWebService(url);
-                    checkResource.check();
+                credURI = localAuthority.getResourceID(Standards.CRED_PROXY_10);
+
+                // The CDP is not applicable to all deployments, so don't throw an exception
+                if (credURI != null) {
+                    URL url = reg.getServiceURL(credURI, Standards.VOSI_AVAILABILITY, AuthMethod.ANON);
+                    if (url != null) {
+                        CheckResource checkResource = new CheckWebService(url);
+                        checkResource.check();
+                    } else {
+                        log.debug("check skipped: " + credURI + " does not provide " + Standards.VOSI_AVAILABILITY);
+                    }
                 } else {
-                    log.debug("check skipped: " + credURI + " does not provide " + Standards.VOSI_AVAILABILITY);
+                    log.debug("check skipped: " + Standards.CRED_PROXY_10 + " not applicable");
                 }
-            } catch (NoSuchElementException ex) {
+            } catch (NoSuchElementException | IllegalArgumentException ex) {
                 log.debug("not configured: " + Standards.CRED_PROXY_10);
             }
 
             URI usersURI = null;
             try {
-                usersURI = localAuthority.getServiceURI(Standards.UMS_USERS_01.toString());
+                usersURI = localAuthority.getResourceID(Standards.UMS_USERS_10);
                 URL url = reg.getServiceURL(credURI, Standards.VOSI_AVAILABILITY, AuthMethod.ANON);
```

Comment:
This looks good, except there is a bug here where it re-uses the possibly null `credURI` rather than `usersURI`... that was already there but probably hidden by the cred check above.